src/overlays/actors/ovl_Bg_Gnd_Darkmeiro/z_bg_gnd_darkmeiro.c

```diff
@@ -35,25 +34,196 @@ const ActorInit Bg_Gnd_Darkmeiro_InitVars = {
+            break;
+    }
+    if (globalCtx) {}; // needed for matching
+}
+
+void BgGndDarkmeiro_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+
+    if ((this->dyna.actor.params & 0xFF) == 1) {
+        if (!globalCtx) {}; // needed for matching
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void BgGndDarkmeiro_DoNothing(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+}
+
+void BgGndDarkmeiro_BlockTimer(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+    s16 timeLeft;
+
+    if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 1) != 0) {
+        if ((this->actionState & 4) != 0) {
+            if (this->timer1 > 0) {
+                this->timer1--;
+            } else {
+                Flags_UnsetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 1);
+                this->actionState &= ~4;
+            }
+        } else {
+            this->actionState |= 4;
+            this->timer1 = 304;
+            Audio_PlaySoundGeneral(NA_SE_EV_RED_EYE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    }
+
+    if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 2) != 0) {
+        if ((this->actionState & 8) != 0) {
+            if (this->timer2 > 0) {
+                this->timer2--;
+            } else {
+                Flags_UnsetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 2);
+                this->actionState &= ~8;
+            }
+        } else {
+            this->actionState |= 8;
+            this->timer2 = 304;
+            Audio_PlaySoundGeneral(NA_SE_EV_RED_EYE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    }
+
+    timeLeft = CLAMP_MIN(this->timer1, this->timer2);
+    if (timeLeft > 0) {
+        func_8002F994(&this->dyna.actor, timeLeft);
+    }
+    if ((this->timer1 >= 64) || (this->timer2 >= 64)) {
+        Flags_SetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F));
+    } else {
+        Flags_UnsetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F));
+    }
+}
+
+void BgGndDarkmeiro_StaticBlock(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+}
+
+void BgGndDarkmeiro_SwitchBlock(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+
+    if (this->timer1 > 0) {
+        this->timer1--;
+    }
+
+    if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F)) != 0) {
+        this->timer1 = 64;
+    }
+
+    BgGndDarkmeiro_ToggleBlock(this, globalCtx);
+}
+
+void BgGndDarkmeiro_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgGndDarkmeiro_Draw0(Actor* thisx, GlobalContext* globalCtx) {
+
+    Gfx_DrawDListXlu(globalCtx, D_060088B0);
+}
+
+void BgGndDarkmeiro_DrawSwitchBlock(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+    s16 vanishTimer;
+    u16 vanishTimerU;
```

Comment:
Have you tried removing this temp? I doubt it's necessary and seems weird for someone to create and use it like it's currently being used.